UpdateProfile/handler.ts

```diff
@@ -62,9 +61,9 @@ export function UpdateProfileHandler(
   return async (context, fiscalCode, profilePayload) => {
     const logPrefix = `UpdateProfileHandler|FISCAL_CODE=${fiscalCode}`;
 
-    const errorOrMaybeExistingProfile = await profileModel.findOneProfileByFiscalCode(
-      fiscalCode
-    );
+    const errorOrMaybeExistingProfile = await profileModel
+      .findLastVersionByModelId([fiscalCode])
```

Comment:
this must be a find() with the modelid compued with the version provided by the client (payload)